localization/yabloc/yabloc_common/include/yabloc_common/ground_server/ground_server.hpp

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef YABLOC_COMMON__GROUND_SERVER__GROUND_SERVER_HPP_
+#define YABLOC_COMMON__GROUND_SERVER__GROUND_SERVER_HPP_
+
+#include "yabloc_common/ground_server/filter/moving_averaging.hpp"
+
+#include <rclcpp/rclcpp.hpp>
+#include <signal_processing/lowpass_filter_1d.hpp>
+#include <yabloc_common/ground_plane.hpp>
+
+#include <autoware_auto_mapping_msgs/msg/had_map_bin.hpp>
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <std_msgs/msg/float32.hpp>
+#include <std_msgs/msg/float32_multi_array.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <visualization_msgs/msg/marker_array.hpp>
+
+#include <lanelet2_core/LaneletMap.h>
+#include <pcl/features/normal_3d.h>
+#include <pcl/kdtree/kdtree_flann.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+namespace yabloc::ground_server
+{
+class GroundServer : public rclcpp::Node
+{
+public:
+  using GroundPlane = common::GroundPlane;
+  using HADMapBin = autoware_auto_mapping_msgs::msg::HADMapBin;
+
+  using Pose = geometry_msgs::msg::Pose;
+  using PoseStamped = geometry_msgs::msg::PoseStamped;
+  using PoseCovStamped = geometry_msgs::msg::PoseWithCovarianceStamped;
+
+  using Float32 = std_msgs::msg::Float32;
+  using Float32Array = std_msgs::msg::Float32MultiArray;  // TODO: define new msg
```

Comment:
Not sure what this TODO is about, but is it OK to leave it as it is?

Reply:
It works without adding new messages, so I will leave it as is. I added a comment explaining what is stored in that message type in d9617f3